util/memory/tracker_test.go

```diff
@@ -82,6 +84,85 @@ func TestConsume(t *testing.T) {
 	require.Equal(t, int64(100), tracker.BytesConsumed())
 }
 
+func TestRelease(t *testing.T) {
+	debug.SetGCPercent(-1)
+	defer debug.SetGCPercent(100)
+	parentTracker := NewGlobalTracker(LabelForGlobalAnalyzeMemory, -1)
+	tracker := NewTracker(1, -1)
+	tracker.AttachToGlobalTracker(parentTracker)
+	require.Equal(t, int64(0), tracker.BytesConsumed())
+	GCAwareMemoryTrack.Store(false)
+	tracker.Consume(100)
+	require.Equal(t, int64(100), tracker.BytesConsumed())
+	require.Equal(t, int64(100), parentTracker.BytesConsumed())
+	tracker.Release(100)
+	require.Equal(t, int64(0), tracker.BytesConsumed())
+	require.Equal(t, int64(0), parentTracker.BytesConsumed())
+	require.Equal(t, int64(0), tracker.BytesReleased())
+	require.Equal(t, int64(0), parentTracker.BytesReleased())
+
+	GCAwareMemoryTrack.Store(true)
+	tracker.Consume(100)
+	require.Equal(t, int64(100), tracker.BytesConsumed())
+	require.Equal(t, int64(100), parentTracker.BytesConsumed())
+	tracker.Release(100)
+	require.Equal(t, int64(0), tracker.BytesConsumed())
+	require.Equal(t, int64(0), parentTracker.BytesConsumed())
+	require.Equal(t, int64(0), tracker.BytesReleased())
+	require.Equal(t, int64(100), parentTracker.BytesReleased())
+	runtime.GC()
+	runtime.GC() // TODO one call cannot work
+	require.Equal(t, int64(0), parentTracker.BytesReleased())
+
+	waitGroup := sync.WaitGroup{}
```

Comment:
use ```util.WaitGroupWrapper``` to avoid using ```Add```/```Done``` by manual.

Reply:
It'll cause an import cycle, need big changes to solve the cycle, let's do it later in another PR.